app/src/debug/java/org/simple/clinic/playground/AwaitActivity.kt

```diff
@@ -0,0 +1,44 @@
+package org.simple.clinic.playground
+
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
+import kotlinx.android.synthetic.debug.await_activity.*
+import org.simple.clinic.R
+import org.simple.clinic.await.Await
+import org.simple.clinic.await.Checkpoint
+import org.simple.clinic.util.unsafeLazy
+
+class AwaitActivity : AppCompatActivity() {
+  private lateinit var awaitDisposable: Disposable
+
+  private val await by unsafeLazy {
+    val checkpoints = listOf(
+        Checkpoint("Wait, one sec...", 0),
+        Checkpoint("Sorry, we literally meant one second!", 2000),
+        Checkpoint("This is taking longer than we expected, it'll take a moment...", 5000),
+        Checkpoint("NVM, it's on us. Please go grab a coffee.", 15000)
+    )
+    Await(checkpoints)
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.await_activity)
+  }
+
+  override fun onStart() {
+    super.onStart()
+    awaitDisposable = await.items()
+        .observeOn(AndroidSchedulers.mainThread())
+        .subscribe { message -> messageTextView.text = message }
+  }
+
+  override fun onStop() {
+    if (::awaitDisposable.isInitialized && awaitDisposable.isDisposed.not()) {
+      awaitDisposable.dispose()
```

Comment:
We can simply call `dispose()` once we make the CompositeDisposable change. We need not check `isDisposed.not()`. It is internally checked inside dispose function of CompositeDisposable.

Reply:
I've changed this to a nullable field. Also, this is not a `CompositeDisposable`

Reply:
Done